client/planning-extension/src/assignments-overview/hiddenAssignmentsList.tsx

```diff
@@ -0,0 +1,141 @@
+import {groupBy, keyBy} from 'lodash';
+import * as React from 'react';
+import {
+    IDesk,
+    ILiveResourcesProps,
+    IRestApiResponse,
+    IUser,
+    IVocabulary,
+} from 'superdesk-api';
+import {IAssignmentItem} from '../../../interfaces';
+import {superdesk} from '../superdesk';
+import {Badge} from 'superdesk-ui-framework';
+import {AssignmentsOverviewListItem} from './assignments-overview-list-item';
+import {getAssignmentsQuery} from '.';
+
+const DropdownTree = superdesk.components.getDropdownTree<IAssignmentItem>();
+const {GroupLabel, IconBig, TopMenuDropdownButton, getLiveQueryHOC, WithLiveResources} = superdesk.components;
+const LiveAssignmentsHOC = getLiveQueryHOC<IAssignmentItem>();
+const {gettext} = superdesk.localization;
+
+interface IState {
+    loading: false;
+    currentUser: IUser;
+}
+
+export class HiddenAssignmentsList extends React.PureComponent<{}, {loading: true} | IState> {
+    constructor(props: {}) {
+        super(props);
+
+        this.state = {loading: true};
+    }
+
+    componentDidMount() {
+        superdesk.session.getCurrentUser()
+            .then((currentUser) => {
+                this.setState({
+                    loading: false,
+                    currentUser: currentUser,
+                });
+            });
+    }
+
+    render() {
+        if (this.state.loading === true) {
+            return null;
+        }
+
+        const {currentUser} = this.state;
+
+        return (
+            <div style={{display: 'none'}}>
+                <LiveAssignmentsHOC resource="assignments" query={getAssignmentsQuery(currentUser._id)}>
+                    {
+                        (data) => {
+                            const assignments = data._items;
+                            const itemsCount = data._meta.total;
+                            const grouped = groupBy(assignments, (item) => item.assigned_to.desk);
+
+
+                            superdesk.dispatchEvent(
+                                'menuItemBadgeValueChange',
+                                {menuId: 'MENU_ITEM_PLANNING_ASSIGNMENTS', badgeValue: itemsCount.toString()},
+                            );
+
+                            const resources: ILiveResourcesProps['resources'] = [
+                                {resource: 'desks', ids: Object.keys(grouped)},
+                                {resource: 'vocabularies', ids: ['g2_content_type']},
+                            ];
+
+                            return (
+                                <WithLiveResources resources={resources}>
```

Comment:
i cant return null bcz of interface... LiveAssignmentsHOC component expect to return another component
ill return empty div then...

Reply:
yep, div is fine too

Reply:
but still i need wrapeer with display none :D

Reply:
but still i need wrapper with display none :D

Reply:
but still i need wrapeer with display none :D

Reply:
![image](https://github.com/user-attachments/assets/65e01805-77c6-4bd9-a61c-2ebb21009387)

adjust the interface so children return not JSX.Element, but React.ReactNode. it will allow null then